hist/hist/src/TGraph2DAsymmErrors.cxx

```diff
@@ -0,0 +1,617 @@
+      for (Int_t i=0; i<GetN(); i++)
+         GetEXlow()[i] *= c1;
+   }
+   if (opt.Contains("x") && GetEXhigh()) {
+      for (Int_t i=0; i<GetN(); i++)
+         GetEXhigh()[i] *= c1;
+   }
+   if (opt.Contains("y") && GetEYlow()) {
+      for (Int_t i=0; i<GetN(); i++)
+         GetEYlow()[i] *= c1;
+   }
+   if (opt.Contains("y") && GetEYhigh()) {
+      for (Int_t i=0; i<GetN(); i++)
+         GetEYhigh()[i] *= c1;
+   }
+   if (opt.Contains("z") && GetEZlow()) {
+      for (Int_t i=0; i<GetN(); i++)
+         GetEZlow()[i] *= c1;
+   }
+   if (opt.Contains("z") && GetEZhigh()) {
+      for (Int_t i=0; i<GetN(); i++)
+         GetEZhigh()[i] *= c1;
+   }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Set number of points in the 2D graph.
+/// Existing coordinates are preserved.
+/// New coordinates above fNpoints are preset to 0.
+
+void TGraph2DAsymmErrors::Set(Int_t n)
+{
+   if (n < 0) n = 0;
+   if (n == fNpoints) return;
+   if (n >  fNpoints) SetPointError(n,0,0,0,0,0,0);
+   fNpoints = n;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Deletes point number ipoint
+
+Int_t TGraph2DAsymmErrors::RemovePoint(Int_t ipoint)
+{
+   if (ipoint < 0) return -1;
+   if (ipoint >= fNpoints) return -1;
+
+   fNpoints--;
+   Double_t *newX      = new Double_t[fNpoints];
+   Double_t *newY      = new Double_t[fNpoints];
+   Double_t *newZ      = new Double_t[fNpoints];
+   Double_t *newEXlow  = new Double_t[fNpoints];
+   Double_t *newEXhigh = new Double_t[fNpoints];
+   Double_t *newEYlow  = new Double_t[fNpoints];
+   Double_t *newEYhigh = new Double_t[fNpoints];
+   Double_t *newEZlow  = new Double_t[fNpoints];
+   Double_t *newEZhigh = new Double_t[fNpoints];
+
+   Int_t j = -1;
+   for (Int_t i = 0; i < fNpoints + 1; i++) {
+      if (i == ipoint) continue;
+      j++;
+      newX[j]      = fX[i];
+      newY[j]      = fY[i];
+      newZ[j]      = fZ[i];
+      newEXlow[j]  = fEXlow[i];
+      newEXhigh[j] = fEXhigh[i];
+      newEYlow[j]  = fEYlow[i];
+      newEYhigh[j] = fEYhigh[i];
+      newEZlow[j]  = fEZlow[i];
+      newEZhigh[j] = fEZhigh[i];
+   }
+   delete [] fX;
+   delete [] fY;
+   delete [] fZ;
+   delete [] fEXlow;
+   delete [] fEXhigh;
+   delete [] fEYlow;
+   delete [] fEYhigh;
+   delete [] fEZlow;
+   delete [] fEZhigh;
+   fX  = newX;
+   fY  = newY;
+   fZ  = newZ;
+   fEXlow  = newEXlow;
+   fEXhigh = newEXhigh;
+   fEYlow  = newEYlow;
+   fEYhigh = newEYhigh;
+   fEZlow  = newEZlow;
+   fEZhigh = newEZhigh;
+   fSize = fNpoints;
+   if (fHistogram) {
+      delete fHistogram;
+      fHistogram = nullptr;
+      fDelaunay = nullptr;
+   }
+   return ipoint;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Set x, y and z values for point number i
+
+void TGraph2DAsymmErrors::SetPoint(Int_t i, Double_t x, Double_t y, Double_t z)
+{
+   if (i < 0) return;
```

Comment:
Olivier, what's the advantage of using the old pattern here?

Reply:
It is not an advantage, but if we want to be coherent we need to change the other TGraph classes too. And my is is that in that case it should another PR.

Reply:
You will have to deal with the change in I/O representation for that hypothetical PR. While here we can have more robust and significantly less code from the beginning - I think I'd value less and better code more than coherent code.

What we can do if you prefer your route is: keep this PR open until you have convinced us that moving TGraph and friends to `vector` is viable, i.e. start the PR you propose before merging this. Would that be an option?

Reply:
Ok fine with me. We can make this new class with std::vector but that will be for the Error vector only because the data part fX fY and fZ are inherited from TGraph2D and are not std::vector . So this class will be mixed any way.
I do not have better arguments. So fine lets go to std::vector for errors only. But here I will need help because I do not know how to use them efficiently.

Reply:
I agree we can keep this open until we move the other TGraph classes to std::vector.